cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -350,6 +350,12 @@ protected ApplicationSpecification regenerateAppSpec(ArtifactDetail artifactDeta
           String.format("No application class found in artifact '%s' in namespace '%s'.",
               artifactDetail.getDescriptor().getArtifactId(), programId.getNamespace()));
     }
+    Map <String, String> runtimeArguments = options.getUserArguments().asMap();
+    if (Boolean.parseBoolean(options.getUserArguments().asMap().getOrDefault(
```

Comment:
So normally we will put this logic in a method in the Arguments class, so it can be reused.

Reply:
ack.